pkg/auth/remoteauthenticator/authenticator.go

```diff
@@ -0,0 +1,172 @@
+package remoteauthenticator
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/auth"
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	"github.com/treeverse/lakefs/pkg/config"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+const DefaultRequestTimeout = 7 * time.Second
+const RemoteAuthSource = "remote_authenticator"
+
+// AuthenticationRequest is the request object that will be sent to the remote authenticator service as JSON payload in a POST request
+type AuthenticationRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// AuthenticationResponse is the expected response from the remote authenticator service
+type AuthenticationResponse struct {
+	// ExternalUserIdentifier is optional, if returned then the user will be used as the official username in lakeFS
+	ExternalUserIdentifier *string `json:"external_user_identifier,omitempty"`
+}
+
+// RemoteAuthenticator client
+type RemoteAuthenticator struct {
+	AuthService auth.Service
+	Logger      logging.Logger
+	Config      *config.RemoteAuthenticator
+	serviceURL  string
+	client      *http.Client
+}
+
+func NewRemoteAuthenticator(conf *config.RemoteAuthenticator, authService auth.Service, logger logging.Logger) (auth.Authenticator, error) {
+	serviceURL, err := url.JoinPath(conf.BaseURL, conf.AuthEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient := http.DefaultClient
+
+	if conf.RequestTimeout == 0 {
+		httpClient.Timeout = DefaultRequestTimeout
+	} else {
+		httpClient.Timeout = conf.RequestTimeout
+	}
```

Comment:
You change the default client if we took a pointer to one and modify the timeout.
If you like to get a default and update the settings - use `httpClient := &http.Client{}`.
In our case we don't need to check the value - just use it:
```go
httpClient := &http.Client{
    Timeout: conf.RequestTimeout,
}
```